src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergChangeEvent.java

```diff
@@ -35,15 +33,17 @@ public class IcebergChangeEvent {
   private final JsonNode value;
   private final JsonNode key;
   private final JsonSchema jsonSchema;
+  private final IcebergSinkConfiguration configuration;
 
   public IcebergChangeEvent(String destination,
                             JsonNode value,
                             JsonNode key,
                             JsonNode valueSchema,
-                            JsonNode keySchema) {
+                            JsonNode keySchema, IcebergSinkConfiguration configuration) {
```

Comment:
I went with what I thought was a more performant solution by injecting class variable state rather than passing the configuration object, as there will be a lot of iteration on the configuration if it is passed in wholesale.  This is definitely cleaner, but adds extra hash table lookups per field processed which impacts performance a bit.   But if I had done this in clojure, I would have passed the configuration everywhere as a map as well.

Reply:
Yeah I don't think this will create a performance bottleneck. The previous solution was drawn to my attention because it caused a unit test to regress due to the mutation of global state.

Which hash table are you referring to? I thought it was just object reference passing and access, which as you say would be replete in a Clojure solution.